classes/security/UserGroupDAO.inc.php

```diff
@@ -458,6 +460,24 @@ public function getByContextId($contextId = null, $dbResultRange = null)
         return new DAOResultFactory($result, $this, '_returnFromRow', [], $sql, $params, $dbResultRange);
     }
 
+    /**
+     * Retrieves a keyed Collection (key = user_group_id, value = count) with the amount of active users for each user group
+     */
+    public function getUserCountByContextId(int $contextId = null): Collection
+    {
+        return DB::table('user_groups', 'ug')
+            ->join('user_user_groups AS uug', 'uug.user_group_id', '=', 'ug.user_group_id')
+            ->join('users AS u', 'u.user_id', '=', 'uug.user_id')
+            ->when($contextId, function (Builder $query) use ($contextId): void {
```

Comment:
I'd like us to use strict comparisons on filters, rather than treating falsy values equivalently. The DAOs are quite sloppy about this but the Collectors are much cleaner and there's no harm in being stricter when introducing/reworking code in DAOs too. So:
```PHP
->when($contextId !== null, function(Builder $query) use ($contextId) {
```

Reply:
Ok!